suite/tests/linux/rseq.c

```diff
@@ -927,6 +1047,119 @@ test_rseq_side_exit(void)
     assert(commits == 0);
 }
 
+/* Tests that DR and drmemtrace handle a side exit in the instrumented sequence. */
+static void
+test_rseq_instru_side_exit(void)
+{
+    volatile struct rseq *reg_rseq = get_my_rseq();
+    int restarts = 0;
+    int commits = 0;
+#ifdef X86
+    __asm__ __volatile__(
+        /* clang-format off */ /* (avoid indenting next few lines) */
+        RSEQ_ADD_TABLE_ENTRY(side_instru, 2f, 3f, 4f)
+        /* clang-format on */
+
+        "6:\n\t"
+        /* Store the entry into the ptr. */
+        "leaq rseq_cs_side(%%rip), %%rax\n\t"
+        "movq %%rax, %[rseq_cs]\n\t"
+        "pxor %%xmm0, %%xmm0\n\t"
+        "mov $1,%%rax\n\t"
+        "movq %%rax, %%xmm1\n\t"
+
+        /* Restartable sequence. */
+        "2:\n\t"
+        /* Increase xmm0 every time.  DR (currently) won't restore xmm inputs
+         * to rseq sequences, nor does it detect that it needs to.
+         */
+        "paddq %%xmm1,%%xmm0\n\t"
+        "movq %%xmm0, %%rax\n\t"
+        /* Take a side exit in the 1st run == instrumented run. */
+        "cmp $2, %%rax\n\t"
```

Comment:
Wouldn't rax be `$1` in the first run?

Reply:
Yes, it's a jne, not a je (b/c copied from the inverted one).  Changed to $1 and je to be more straightforward.